packages/iTwinUI-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -45,9 +52,9 @@ export type ComboBoxProps<T> = {
    * Function to customize the default filtering logic.
    */
   filterFunction?: (
-    options: SelectOption<T>[],
+    options: ComboBoxOption<T>[],
```

Comment:
Isnt this one a breaking change????

Reply:
Is it? This is a function we call. The user just takes whatever from the `options`.

Reply:
In my understanding it is the change of API (types).
Not only the coming argument type changed (so if users defined that explicitly, will get errors), but also the return type changed (if users do not do `..rest` they will end up with errors too ) :/

Reply:
Oh yeah, good point about return type. Hmm, I will revert this then

Reply:
This is going to be tough because our `filteredOptions` state expects `ComboBoxOption` so that it is able to access `__originalIndex`

Reply:
What can we do?

Reply:
Lets fix this and I approve. Too long hanging.

Reply:
Fixed by saving `__originalIndex` in `optionsExtraInfoRef` and getting rid of `ComboBoxOption`